### ISSUE TITLE
iOS default Start

### ISSUE THREAD

**Issue by SpatialStrout** [dpa99c/phonegap-launch-navigator#10]
Thanks for the cool plugin. 
Your doc says that leaving the start parameter as null should default to the current location as the start but my implementation is pulling up the directions and the user has to select the Start (Current Location is one of the options). I found that if I use "Current Location" as the start parameter, it will do this for me but not if I just leave it null. I haven't tested this on other platforms yet.

**Comment by dpa99c** [comment]
What version of iOS are you testing on? Do you get the same behaviour with the [example project](https://github.com/dpa99c/phonegap-launch-navigator-example)?

**Comment by kcaran** [comment]
I didn't try the example project, but I'm seeing the same on iOS 8.2. This occurs even if I use the maps://daddr=Address or maps://q=Address url directly (without the plugin).

**Comment by dpa99c** [comment]
I've done some testing and found that while passing null as the start location works on iOS 8.0 and 8.1, it doesn't work with iOS 8.2 and 8.3.
It seems there was a change in iOS 8.2 which means passing null location no longer works. I'll update the plugin code, examples and documentation to reflect this.

**Comment by dpa99c** [comment]
This issue should now be fixed in v2.2.0 of the plugin; if you have any issues, please test the example project first, then let me know.

**Comment by DePaMa** [comment]
Hey dpa99c,
are you sure this issue is resolved? I have some issues with an iPhone5 iOS 8.4.1. In my ionic app, i always get this error with an app crash: 
```
2015-09-14 12:16:31.610 MyApp[40841:149551] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSPlaceholderArray initWithObjects:count:]: attempt to insert nil object from objects[1]'
*** First throw call stack:
(
	0   CoreFoundation                      0x0028f746 __exceptionPreprocess + 182
	1   libobjc.A.dylib                     0x026a4a97 objc_exception_throw + 44
	2   CoreFoundation                      0x0013cff5 -[__NSPlaceholderArray initWithObjects:count:] + 405
	3   CoreFoundation                      0x001a03a4 +[NSArray arrayWithObjects:count:] + 68
	4   MyApp                     0x0006a50e -[LaunchNavigator openAppleMaps:] + 4318
	5   MyApp                     0x00068917 -[LaunchNavigator navigate:] + 935
	6   MyApp                     0x00083ec8 -[CDVCommandQueue execute:] + 888
	7   MyApp                     0x00083700 -[CDVCommandQueue executePending] + 848
	8   MyApp                     0x00084e53 -[CDVCommandDelegateImpl evalJsHelper2:] + 355
	9   libobjc.A.dylib                     0x026ba771 -[NSObject performSelector:withObject:] + 70
	10  Foundation                          0x022a14e8 __NSThreadPerformPerform + 330
	11  CoreFoundation                      0x001b106f __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 15
	12  CoreFoundation                      0x001a6b7d __CFRunLoopDoSources0 + 253
	13  CoreFoundation                      0x001a60d8 __CFRunLoopRun + 952
	14  CoreFoundation                      0x001a5a5b CFRunLoopRunSpecific + 443
	15  CoreFoundation                      0x001a588b CFRunLoopRunInMode + 123
	16  GraphicsServices                    0x03c382c9 GSEventRunModal + 192
	17  GraphicsServices                    0x03c38106 GSEventRun + 104
	18  UIKit                               0x0064e0b6 UIApplicationMain + 1526
	19  MyApp                     0x0004c3a9 main + 89
	20  libdyld.dylib                       0x02d61ac9 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) 
```

so i tried your example app, and i get this error:
```
2015-09-14 12:21:14.289 LaunchNavigatorExample[40916:151245] [objc]: Debug mode enabled
2015-09-14 12:21:14.291 LaunchNavigatorExample[40916:151245] [objc]: Launching Apple Maps...
2015-09-14 12:21:14.293 LaunchNavigatorExample[40916:151245] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSPlaceholderArray initWithObjects:count:]: attempt to insert nil object from objects[1]'
*** First throw call stack:
(
	0   CoreFoundation                      0x00242746 __exceptionPreprocess + 182
	1   libobjc.A.dylib                     0x02657a97 objc_exception_throw + 44
	2   CoreFoundation                      0x000efff5 -[__NSPlaceholderArray initWithObjects:count:] + 405
	3   CoreFoundation                      0x001533a4 +[NSArray arrayWithObjects:count:] + 68
	4   LaunchNavigatorExample              0x0001b7fe -[LaunchNavigator openAppleMaps:] + 4318
	5   LaunchNavigatorExample              0x00019c07 -[LaunchNavigator navigate:] + 935
	6   LaunchNavigatorExample              0x0003e9d8 -[CDVCommandQueue execute:] + 888
	7   LaunchNavigatorExample              0x0003e210 -[CDVCommandQueue executePending] + 848
	8   LaunchNavigatorExample              0x0003f963 -[CDVCommandDelegateImpl evalJsHelper2:] + 355
	9   libobjc.A.dylib                     0x0266d771 -[NSObject performSelector:withObject:] + 70
	10  Foundation                          0x022544e8 __NSThreadPerformPerform + 330
	11  CoreFoundation                      0x0016406f __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 15
	12  CoreFoundation                      0x00159b7d __CFRunLoopDoSources0 + 253
	13  CoreFoundation                      0x001590d8 __CFRunLoopRun + 952
	14  CoreFoundation                      0x00158a5b CFRunLoopRunSpecific + 443
	15  CoreFoundation                      0x0015888b CFRunLoopRunInMode + 123
	16  GraphicsServices                    0x03beb2c9 GSEventRunModal + 192
	17  GraphicsServices                    0x03beb106 GSEventRun + 104
	18  UIKit                               0x006010b6 UIApplicationMain + 1526
	19  LaunchNavigatorExample              0x000184f9 main + 89
	20  libdyld.dylib                       0x02d14ac9 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) 
```
so it seems to be the same issue. 

I tried to navigate with and without "start" param, but the error is still the same. Startparam was a string and also an array with coordinates. I tried the example on an Iphone5 device and the MacOs Emulator on my MacMini, everytime the same issue.

I am sure that it worked before the iOS update but I am not sure when it gets broken.

Regards,

DePaMa

**Comment by dpa99c** [comment]
@DePaMa can you check you are using the latest version of the plugin? It should be v2.9.5

I'm not able to reproduce the errors you are seeing when building and running the example app with this version of the plugin.

Can you try to download the [pre-built IPA](https://github.com/dpa99c/phonegap-launch-navigator-example/blob/master/build/LaunchNavigatorExample.ipa) for the example project and install via iTunes? This is built on my local machine, signed with my iOS Distribution Certificate and works fine on my iPad 2 running iOS 8.3

Do any examples in the example app work or all they all broken for you?

**Comment by DePaMa** [comment]
Hey dpa99c,
yes it works now! Thanks! I don't know why but cordova always added the version 2.9.2. I copied the version 2.9.5 to my folder and everything is fine now! Think can be closed again :)

**Comment by dpa99c** [comment]
Good to hear it's working. 

Bear in mind that if your Cordova CLI verison is less than 5.0.0, it will use the Cordova Plugins registry which is now deprecated and read-only (plugins cannot be updated). For this reason, the last version of this plugin published to there was [2.9.2](http://plugins.cordova.io/#/package/uk.co.workingedge.phonegap.plugin.launchnavigator).

Make sure your Cordova CLI is updated to latest version - `npm install -g cordova`. This will then find cordova plugins on npmjs, where the latest version of the plugin in [2.9.5](https://www.npmjs.com/package/uk.co.workingedge.phonegap.plugin.launchnavigator)
